### PR TITLE
feat(dumper): dump js mongodb models

### PR DESCRIPTION
**Pull request by SteveBunlon**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by SteveBunlon** [comment]
This PR contains additional works that needs to be taken out: 
- move lodash in the context
- strings utils is now in ts format inject in the context

This two PRs should be created after [this one](https://github.com/ForestAdmin/toolbelt/pull/524) has been merged

File: src/services/dumpers/agent-nodejs.ts [hunk 1]
```diff
@@ -61,6 +67,7 @@ export default class AgentNodeJs extends AbstractDumper {
 
     if (dbDialect === 'mongodb') {
       dependencies['@forestadmin/datasource-mongoose'] = '^1.0.0';
+      dependencies.mongoose = '^6.8.3';
```
Comment (jeffladiray) [inline src/services/dumpers/agent-nodejs.ts:70]: Same as the previous one I guess, maybe we could upgrade to the latest 6.x ?

File: src/services/dumpers/agent-nodejs.ts [hunk 2]
```diff
@@ -108,19 +115,23 @@ export default class AgentNodeJs extends AbstractDumper {
...
-      datasourceCreation: isMongoose
-        ? 'createMongooseDataSource(connection, {})'
-        : `
+      datasourceImport: null,
+      datasourceCreation: null,
+    };
+
+    if (isMongoose) {
+      context.datasourceImport = `const { createMongooseDataSource } = require('@forestadmin/datasource-mongoose');\nconst primaryConnection = require('./models/primary');`;
+      context.datasourceCreation = `createMongooseDataSource(primaryConnection, { flattenMode: 'auto' })`;
```
Comment (jeffladiray) [inline src/services/dumpers/agent-nodejs.ts:124]: So, models will be written in `./models/primary`? Any reason for the `primary`naming ?

File: src/services/dumpers/agent-nodejs.ts [hunk 3]
```diff
@@ -191,7 +202,38 @@ export default class AgentNodeJs extends AbstractDumper {
...
+    });
+  }
+
+  private async writeMongooseModels(schema) {
+    await this.mkdirp(`${this.projectPath}/models/primary`);
+
+    this.copyHandleBarsTemplate('models/index.hbs', 'models/primary/index.js');
+
+    this.writeModels(schema);
+  }
```
Comment (jeffladiray) [inline src/services/dumpers/agent-nodejs.ts:234]: I think we should merge both function.

`writeModels` is mongoose specific atm, and `writeMongooseModels` does almost nothing outside creating the folder and copy index.

File: src/services/dumpers/agent-nodejs.ts [hunk 4]
```diff
@@ -205,5 +247,9 @@ export default class AgentNodeJs extends AbstractDumper {
     this.writeDockerignore();
     this.writeDockerfile();
     this.writeDockerCompose(dumpConfig);
+
+    if (schema) {
```
Comment (jeffladiray) [inline src/services/dumpers/agent-nodejs.ts:251]:
```suggestion
    if (isMongoose) {
```
(I didn't test obviously, but this if(schema) -> writeMongoose seems weird to me)

**Review by jeffladiray — COMMENTED** [review]
Same here, didn't review the tests.

These are pretty hard to read, I prefer to manually check that everyone of them fails if I try to modify the expected result.

File: src/services/dumpers/agent-nodejs.ts (hunk 2 above)
Comment (SteveBunlon) [reply src/services/dumpers/agent-nodejs.ts:120]: Nop. I've just added the structure to add multi-database support by default. Primary is basically a name, could be `default`, `first-one`, even your name if you want.

File: src/services/dumpers/agent-nodejs.ts (hunk 3 above)
Comment (SteveBunlon) [reply src/services/dumpers/agent-nodejs.ts:228]: Ok let's go.

File: src/services/dumpers/agent-nodejs.ts (hunk 4 above)
Comment (SteveBunlon) [reply src/services/dumpers/agent-nodejs.ts:241]: I update bot the if statement and the schema variable name.

**Pull request merged**